rs/registry/canister/canister/canister.rs

```diff
@@ -185,60 +185,67 @@ ic_nervous_system_common_build_metadata::define_get_build_metadata_candid_method
 
 #[export_name = "canister_query get_changes_since"]
 fn get_changes_since() {
-    // Only self-authenticating and anonymous principals are allowed to call us.
-    const ALLOWED_CALLER_CLASSES: [Result<PrincipalIdClass, String>; 2] = [
-        Ok(PrincipalIdClass::SelfAuthenticating),
-        Ok(PrincipalIdClass::Anonymous),
-    ];
-    let class = caller().class();
-    if !ALLOWED_CALLER_CLASSES.contains(&class) {
-        let response = RegistryGetChangesSinceResponse {
-            error: Some(RegistryError {
-                code: Code::Authorization as i32,
-                reason: format!(
+    fn main() -> Result<RegistryGetChangesSinceResponse, (Code, String)> {
+        // Authorize: Only self-authenticating and anonymous principals are allowed to call us.
+        const ALLOWED_CALLER_CLASSES: [Result<PrincipalIdClass, String>; 2] = [
+            Ok(PrincipalIdClass::SelfAuthenticating),
+            Ok(PrincipalIdClass::Anonymous),
+        ];
+        let class = caller().class();
+        if !ALLOWED_CALLER_CLASSES.contains(&class) {
+            return Err((
+                Code::Authorization,
+                format!(
                     "Caller must be self-authenticating, or anonymous (but was {:?}).",
                     class,
                 ),
-                key: vec![],
-            }),
-            ..Default::default()
```

Comment:
This is useless, and repeated later. Getting rid of this is one of the minor improvements in this PR.